#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/GlossyPaint.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/MattePaint.proto"

PROTO RobotSaver [
    field SFInt32       id              666
    field SFString      name            "robot_saver"
    field SFVec3f       translation     5 5 1
    field SFRotation    rotation        0 0 1 0
    field SFVec3f       scale           0.001 0.001 0.001
    field SFString      controller      "test"
    field SFBool        supervisor       FALSE
]
{
    Robot {
        name IS name
        controller IS controller
        supervisor IS supervisor

        translation IS translation
        rotation IS rotation

        children [
            DEF LEFT_MOTOR Propeller {
                shaftAxis          -1 0 0
                centerOfThrust     0 -3 0
                thrustConstants    100 0
                torqueConstants    100 0
                fastHelixThreshold 75

                device RotationalMotor {
                    name "robot_left_motor"
                    maxTorque 200
                }

                fastHelix NULL
                slowHelix NULL
                
            }

            DEF LEFT_MOTOR Propeller {
                shaftAxis          -1 0 0
                centerOfThrust     0 3 0
                thrustConstants    100 0
                torqueConstants    100 0
                fastHelixThreshold 75

                device RotationalMotor {
                    name "robot_right_motor"
                    maxTorque 200
                }

                fastHelix NULL
                slowHelix NULL
                
            }

            DEF GYRO Gyro {
            }

            DEF INERTIAL_UNIT InertialUnit {
            }

            DEF RECEIVER Receiver {
                type "serial"
            }

            Solid {
                

                physics Physics {
                    density 20
                    mass -1
                    
                }

                boundingObject Pose {
                    translation 0 0 0.5

                    children [
                        Box {
                            size 4.8 2.8 1
                        }
                    ]
                }
            
                children [
                   Transform {
                        scale IS scale

                        children [
                            Shape {
                                appearance PBRAppearance {
                                    baseColor 1 0 0
                                }

                                geometry Mesh {
                                    url "../models/robot.dae"
                                }
                            }
                        ]
                   }

                ]
            }
        ]

        physics Physics {
            density -1
            mass 50
            inertiaMatrix [1 1 1 0 0 0]
            centerOfMass [ -0.088 0 -0.04]
        }
    }
}